code/Model/DynamoDbSession.php

```diff
@@ -64,10 +50,7 @@ public static function get()
                 $dynamoOptions['credentials']['key'] = $awsAccessKey;
                 $dynamoOptions['credentials']['secret'] = $awsSecretKey;
             } else {
-                // cache credentials when IAM fetches the credentials from EC2 metadata service
-                // this will use doctrine/cache (included via composer) to do the actual caching into APCu
-                // http://docs.aws.amazon.com/aws-sdk-php/guide/latest/performance.html#cache-instance-profile-credentials
-                $dynamoOptions['credentials'] = new DoctrineCacheAdapter(new ApcuCache());
+                $dynamoOptions['credentials'] = CredentialProvider::defaultProvider();
```

Comment:
[doctrine/cache](https://github.com/doctrine/cache) is deprecated - this is from their readme:
> This library is deprecated and will no longer receive bug fixes from the Doctrine Project. Please use a different cache library, preferably PSR-6 or PSR-16 instead.

The default credential provider is memoised, which is a fancy way of saying it caches the credentials after finding them.